ntrans_gui.py

```diff
@@ -180,22 +208,42 @@ def generate_ntrans_dictionary(self) -> None:
             "target_language": self.target_language_var.get(),
         }
 
-        for key in self.user_choices:
-            if not self.user_choices[key]:
+        for key in user_choices:
+            if not user_choices[key]:
                 print(
                     "All fields need to be filled in before generating the N-Trans dictionary."
                 )  # TODO: Change print to label
                 break
 
         # TODO: Check filepath for validity
+        if not hasattr(ntrans_gui, "thread"):
+            self.progress_queue: queue.Queue[int] = queue.Queue()
+            # Calls logic in ntrans.py
+            self.thread = threading.Thread(
+                target=ntrans.read_ngram_files, args=[user_choices, self.progress_queue]
+            )
+            self.thread.start()
 
-        # Calls logic in ntrans.py
-        ntrans.read_ngram_files(self.user_choices)
+        self.open_progress_bar()
 
-    def open_help_page(self) -> None:
-        webbrowser.open_new_tab(
-            "https://www.google.com"
-        )  # TODO: Write help document and link to it
+    def open_progress_bar(self) -> None:
+        self.estimated_time_label.grid_remove()
+        self.progress_frame.grid(
+            column=0, row=16, columnspan=2, padx=(0, 0), pady=(20, 40)
+        )
+        self.check_progressbar_queue()
```

Comment:
I wouldn't keep this in a separate method, since it's only called in one place and there's not that much code.